alicloud/resource_alicloud_launch_template.go

```diff
@@ -0,0 +1,490 @@
+	args.IoOptimized = d.Get("io_optimized").(string)
+	args.KeyPairName = d.Get("key_pair_name").(string)
+	args.NetworkType = d.Get("network_type").(string)
+
+	args.RamRoleName = d.Get("ram_role_name").(string)
+	args.ResourceGroupId = d.Get("resource_group_id").(string)
+	args.SecurityEnhancementStrategy = d.Get("security_enhancement_strategy").(string)
+	args.SecurityGroupId = d.Get("security_group_id").(string)
+	args.SpotPriceLimit = requests.NewFloat(d.Get("spot_price_limit").(float64))
+	args.SpotStrategy = d.Get("spot_strategy").(string)
+	args.SystemDiskDiskName = d.Get("system_disk_name").(string)
+	args.SystemDiskCategory = d.Get("system_disk_category").(string)
+	args.SystemDiskDescription = d.Get("system_disk_description").(string)
+	args.SystemDiskSize = requests.NewInteger(d.Get("system_disk_size").(int))
+	args.UserData = d.Get("userdata").(string)
+	args.VSwitchId = d.Get("vswitch_id").(string)
+	args.VpcId = d.Get("vpc_id").(string)
+	args.ZoneId = d.Get("zone_id").(string)
+	netsRaw := d.Get("network_interfaces").([]interface{})
+	if netsRaw != nil {
+		var nets []ecs.CreateLaunchTemplateNetworkInterface
+		for _, raw := range netsRaw {
+			netRaw := raw.(map[string]interface{})
+			net := ecs.CreateLaunchTemplateNetworkInterface{
+				NetworkInterfaceName: netRaw["name"].(string),
+				VSwitchId:            netRaw["vswitch_id"].(string),
+				SecurityGroupId:      netRaw["security_group_id"].(string),
+				Description:          netRaw["description"].(string),
+				PrimaryIpAddress:     netRaw["primary_ip"].(string),
+			}
+			nets = append(nets, net)
+		}
+		args.NetworkInterface = &nets
+	}
+
+	disksRaw := d.Get("data_disks").([]interface{})
+	if disksRaw != nil {
+		var disks []ecs.CreateLaunchTemplateDataDisk
+		for _, raw := range disksRaw {
+			diskRaw := raw.(map[string]interface{})
+			disk := ecs.CreateLaunchTemplateDataDisk{
+				Size:               fmt.Sprintf("%d", diskRaw["size"].(int)),
+				SnapshotId:         diskRaw["snapshot_id"].(string),
+				Category:           diskRaw["category"].(string),
+				Encrypted:          fmt.Sprintf("%v", diskRaw["encrypted"].(bool)),
+				DiskName:           diskRaw["name"].(string),
+				Description:        diskRaw["description"].(string),
+				DeleteWithInstance: fmt.Sprintf("%v", diskRaw["delete_with_instance"].(bool)),
+			}
+			disks = append(disks, disk)
+		}
+
+		args.DataDisk = &disks
+	}
+	tagsRaw := d.Get("tags").(map[string]interface{})
+	var tags []ecs.CreateLaunchTemplateTag
+	for key, value := range tagsRaw {
+		tags = append(tags, ecs.CreateLaunchTemplateTag{
+			Key:   key,
+			Value: value.(string),
+		})
+	}
+	args.Tag = &tags
+
+	raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.CreateLaunchTemplate(args)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "launch_template", args.GetActionName(), AlibabaCloudSdkGoERROR)
```

Comment:
Please modify line 379 to `return WrapErrorf(err, DefaultErrorMsg, "alicloud_launch_template", args.GetActionName(), AlibabaCloudSdkGoERROR)`